src/jmcomic_ai/core.py

```
@@ -302,51 +307,246 @@ def get_ranking(self, period: str = "day", page: int = 1) -> dict[str, Any]:
+                super().before_photo(photo)
+                self.total = len(photo)
+
+                import json
+                photo_dict = {
+                    "id": str(photo.photo_id),
+                    "name": str(photo.name),
+                    "total_pages": self.total,
+                }
+                msg = f"📖 Photo Info: {json.dumps(photo_dict, ensure_ascii=False)}"
+                service_logger.info(msg)
+                service._safe_ctx_call(ctx, loop, lambda: ctx.info(msg), "Failed to send photo info to ctx")
```

Ruff (0.14.13)

Undefined name `service`

(F821)